tools/syz-reprolist/reprolist.go

```diff
@@ -28,10 +32,111 @@ var (
 	flagOutputDir    = flag.String("output", "repros", "output dir")
 	flagSyzkallerDir = flag.String("syzkaller", ".", "syzkaller dir")
 	flagOS           = flag.String("os", runtime.GOOS, "target OS")
+	flagNamespace    = flag.String("namespace", "", "target namespace")
+	flagToken        = flag.String("token", "", "gcp bearer token to disable throttling (contact syzbot first)\n"+
+		"usage example: ./tools/syz-reprolist -namespace upstream -token $(gcloud auth pring-access-token)")
 )
 
+func getJSONBody(url string) ([]byte, error) {
+	if strings.Contains(url, "?") {
+		url = url + "&json=1"
+	} else {
+		url = url + "?json=1"
+	}
+	req, err := http.NewRequest("GET", url, nil)
+	if *flagToken != "" {
+		req.Header.Add("Authorization", "Bearer "+*flagToken)
+	} else {
+		time.Sleep(time.Second) // tolerate throttling
+	}
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("http.Get(%s): %w", url, err)
+	}
+	defer res.Body.Close()
+	body, err := io.ReadAll(res.Body)
+	if res.StatusCode > 299 {
+		return nil, fmt.Errorf("io.ReadAll failed with status code: %d and\nbody: %s", res.StatusCode, body)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("io.ReadAll(body): %s", err.Error())
+	}
+	return body, nil
+}
+
+func saveRepro(bugID, reproID string, reproData []byte) error {
+	bugDir := path.Join(*flagOutputDir, bugID)
+	if err := os.MkdirAll(bugDir, 0777); err != nil {
+		return fmt.Errorf("os.MkdirAll: %w", err)
+	}
+	reproPath := path.Join(bugDir, reproID)
+	if err := os.WriteFile(reproPath, reproData, 0666); err != nil {
```

Comment:
This will be 1 file per dir. Looks like dirs are not really needed here, will just make any later processing more complex.
We could save just `bugID.c` files.
Also won't need mkdir.

Reply:
Hard to answer these questions w/o seeing/understanding the end destination.
E.g. how metadata will be used, how much of the metadata we need, do we want to duplicate it the repro archive/repo, or we want to refer to the data for meta information, what backwards compatibility implications the export will have, how these reproducers will run, will it be possible to be export all, but run just one per bug on demand, etc.

Reply:
So I am just looking only at what we have in this PR and commenting with respect to that.